DependencyInjection/Configuration.php

```diff
@@ -69,6 +69,26 @@ public function getConfigTreeBuilder()
                     return $v;
                 })
             ->end()
+            ->beforeNormalization() // @deprecated toolbar in 1.3.0, use profiling instead
+                ->ifTrue(function ($v) {
+                    return is_array($v) && array_key_exists('toolbar', $v) && is_array($v['toolbar']);
+                })
+                ->then(function ($v) {
+                    if (array_key_exists('enabled', $v['toolbar']) && 'auto' === $v['toolbar']['enabled']) {
+                        $v['toolbar']['enabled'] = $this->debug; // @deprecated value auto in 1.3.0
+                    }
+
+                    if (array_key_exists('profiling', $v) && is_array($v['profiling'])) {
```

Comment:
I think the BC promise at this point is that toolbar will work at least until 2.0 which means if toolbar is configured it should always work, otherwise it's a BC break, no matter if an error is thrown or overwritten by profiling.

What about default values? you mean the auto config? I added it here so that the enabled config can be defined as boolean.